### PR TITLE
fix: CE-608 Animal drug date required when auto-fill

### PR DESCRIPTION
**Pull request by Scarlett-Truong**
---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-compliance-enforcement-343-backend.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://nr-compliance-enforcement-343-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/app/store/reducers/case-thunks.ts
Comment (Scarlett-Truong) [inline frontend/src/app/store/reducers/case-thunks.ts:1]: This change is just to add back the missing code that have been merged in [this PR](https://github.com/bcgov/nr-compliance-enforcement/pull/329/files#diff-8f66c21cca11dfdeec23a85d4fb894b0cc0c2a33567f39edd8ae9c79741084c0).

**Review by dmitri-korin-bcps — COMMENTED** [review]
Please remove unnecessary useEffect from the first line of drug-authorization.tsx. Other than that looks good.

**Comment by afwilcox** [comment]
@Scarlett-Truong - looks like your latest change pushed that function above the complexity threshold.   Are you able to refactor it to make it less complex?  (For example, check for nulls up to remove nesting?)

**Comment by Scarlett-Truong** [comment]
Open new one at https://github.com/bcgov/nr-compliance-enforcement/pull/370

**Pull request closed without merging**